flusher.go

```diff
@@ -19,57 +19,122 @@ import (
 // Flush takes the slices of metrics, combines then and marshals them to json
 // for posting to Datadog.
 func (s *Server) Flush(interval time.Duration, metricLimit int) {
+	// right now we have only one destination plugin
+	// but eventually, this is where we would loop over our supported
+	// destinations
+	if s.IsLocal() {
+		s.FlushLocal(interval, metricLimit)
+	} else {
+		s.FlushGlobal(interval, metricLimit)
+	}
+}
+
+func (s *Server) FlushGlobal(interval time.Duration, metricLimit int) {
 	go s.flushEventsChecks() // we can do all of this separately
 
 	percentiles := s.HistogramPercentiles
-	if s.ForwardAddr != "" {
-		// don't publish percentiles if we're a local veneur; that's the global
-		// veneur's job
-		percentiles = nil
-	}
 
+	tempMetrics, ms := s.tallyMetrics(percentiles)
+
+	// the global veneur instance is also responsible for reporting the sets
+	ms.totalLength += ms.totalSets
+
+	finalMetrics := s.generateDDMetrics(interval, percentiles, tempMetrics, ms)
+
+	s.reportMetricsFlushCounts(ms)
+
+	s.reportGlobalMetricsFlushCounts(ms)
+
+	s.flushRemote(finalMetrics, metricLimit)
```

Comment:
okay so in the long run we intend for flushRemote to potentially be generalized across remote targets, or even dealing with multiple remote targets at a time? makes sense to me